### DISCUSSION TITLE
Python multipart patch to support projects using FastAPI <=0.9XX

### DISCUSSION THREAD

**Discussion by jackirvine97** [fastapi/fastapi#11252] [Questions] [answered]
### Description

**Context:** Fast API 0.109.1 released a patch for [GHSA-qf9m-vfgh-m389](https://github.com/tiangolo/fastapi/security/advisories/GHSA-qf9m-vfgh-m389). This is a High sev vulnerability related to python-multipart<0.0.7

**Issue:** this fix requires an upgrade to pydantic v2. This is a large effort for existing projects and not something that can be done quickly. This means projects with pydantic v1 and fast API <0.9XX are exposed to this vulnerability. This set-up remains common given the widespread production use of fastAPI and the fact that the move to V2 took place only recently

**Ask:** could 0.99.1 be patched to provide cover for projects using a pydantic V1 set-up?

### Operating System

Linux

### FastAPI Version

0.99.1

### Pydantic Version

1.10

### Python Version

3.11

**Comment by Kludex** [comment]
FastAPI is compatible with Pydantic v1 and v2.

**Reply by jackirvine97** [reply]
Thanks for getting back to me @Kludex! In the case I outline above, I have a large application which uses 0.99.1 and pydantic V1. If i upgrade fast API to 0.110, the APIs break as they cannot handle being passed V1 pydantic models.

Here is a contrived example of what is breaking:

```
@router.get("/my_api)
async def my_api(
   body: SomePydanticV1Model,
):
...
```
Other V1 only features such as `conlist` seem to cause issues too

**Reply by jackirvine97** [reply]
The ultimate end goal is to upgrade the models to V2 - however, in a large application /org this is not a quick win sadly (as it requires funding, testing etc). Given I suspect there will be others still using a V1 pydantic + fast api <0.9XX position (as it up until recently was the norm) , I wondered if there is any possibility of getting a patch to the older versions to support these users so they are not exposed to the vulnerability (whilst in the longer term process of migrating)?

**Reply by tiangolo** [reply]
Apart from the original matter (that @Kludex already answered :bow:), @jackirvine97 it would be great if you could create a minimal example of code that I can copy-paste with the latest FastAPI and Pydantic v 1.10.x that doesn't work. Because it should work... :sweat_smile:

**Comment by Kludex** [comment, marked as answer]
Hi... Sorry. I just remembered something important.

The security advisory on FastAPI was unnecessary. The issue was on the `python-multipart` optional dependency. The thing is that we didn't have a way to publish an advisory on `python-multipart` when that happened, because the project was under a GitHub user that is not that much active anymore. Now I (Kludex) have the ownership of `python-multipart`, but the advisory on FastAPI cannot be undone because GitHub doesn't have that feature yet.

See what GitHub has to say about it [here](https://github.com/github/advisory-database/issues/3085#issuecomment-1933049782).

Anyway, it's safe to not bump FastAPI, but bump `python-multipart` on your project.

**Reply by jackirvine97** [reply]
Thanks @Kludex - that is a very helpful answer 

We also chimed onto the optional nature of this - in fact, when installing ‘fastapi’ with pipenv, it doesn’t install multipart at all (you have to install ‘fastapi[all]’ to pick it up. So in short, basic security package checks were false flagging!!

thanks for your help!!

**Reply by Kludex** [reply]
You are welcome. 🙏
